src/validate.ts

```diff
@@ -0,0 +1,75 @@
+
+'use-strict';
+
+
+function generateRequestID(min, max){
+        return Math.floor(Math.random() * (max-min) + min).toString()
+}
+    
+
+export function generatSyncRequest(){
+    const requestId = generateRequestID(100,999);
+    return {
+        requestId,
+        inputs: [{
+          intent: "action.devices.SYNC"
+        }]
+    }
+}
+
+
+let deviceIDs: Array<string>;
+
+interface customDatas = {
+    [key: string]: string | number | boolean
+}
+
+export function generateQueryRequest(deviceIds, customDatas){
```

Comment:
It seems like these functions do not have typed parameters or return types.

You can specify these typings inline following a colon:

```typescript
export function generateQueryRequest(deviceIds: string[], customDatas: customDatas[]): QueryRequest
```

There is no QueryRequest type that exists right now, so that would need to be defined.

Reply:
Pushed these changes, I'm not entirely positive if the way I declared the QueryRequest was correct. I feel like more variables are needed but lmk.

Reply:
Yeah this follows the [QUERY request format](https://developers.google.com/assistant/smarthome/reference/intent/query).